src/components/ConnectToDB/ConnectToDBDialog.tsx

```diff
@@ -32,9 +37,26 @@ interface ConnectToDBDialogProps extends SnippetParams {
     onClose: VoidFunction;
 }
 
-function ConnectToDBDialog({open, onClose, database, endpoint}: ConnectToDBDialogProps) {
+function ConnectToDBDialog({
+    open,
+    onClose,
+    database,
+    endpoint: endpointFromProps,
+}: ConnectToDBDialogProps) {
     const [activeTab, setActiveTab] = React.useState<SnippetLanguage>('bash');
 
+    const clusterName = useClusterNameFromQuery();
+    const singleClusterMode = useTypedSelector((state) => state.singleClusterMode);
+
+    // If there is enpdoint from props, we don't need to request tenant data
+    // Also we should not request tenant data if we are in single cluster mode
+    // Since there is no ControlPlane data in this case
+    const shouldRequestTenantData = database && !endpointFromProps && !singleClusterMode;
```

Comment:
I left `endpoint` in props to make possible to use dialog in Databases table, where we know all the params from table data and do not need to request them additionally